mocked-api/handlers.ts

```diff
@@ -180,36 +210,278 @@ export const handlers = [
+            bridgeId as string
+          }' for customer 'XXXXXXXX' does not exist`,
+        })
+      );
+    }
+
+    const existingProcessor = db.processor.findFirst({
+      where: {
+        name: {
+          equals: name,
+        },
+        bridge: {
+          id: {
+            equals: bridgeId as string,
+          },
+        },
+      },
+    });
+
+    if (existingProcessor) {
+      return res(
+        ctx.status(400),
+        ctx.json({
+          ...error_duplicated_resource,
+          reason: `Processor with name '${name}' already exists for bridge with id ${
+            bridgeId as string
+          } for customer with id 'XXXXXXXXXX'`,
+        })
+      );
+    }
+
+    const id = uuid();
+    const processor = {
+      kind: "Processor",
+      id,
+      name,
+      href: `/api/v1/bridges/${bridge?.id ?? ""}/processors/${id}`,
+      submitted_at: new Date().toISOString(),
+      status: "accepted",
+      filters: filters,
+      transformationTemplate,
+      action,
+      bridge,
+    };
+
+    const newProcessor = db.processor.create(processor);
+
+    // make the process slower if the resource name contains "wait" and make it fail
+    // if the name contains "fail"
+    resourceStatusFlow(
+      "processor",
+      id,
+      name.includes("wait"),
+      name.includes("fail")
+    );
+
+    return res(ctx.status(200), ctx.delay(apiDelay), ctx.json(newProcessor));
+  }),
+  // delete a processor
+  rest.delete(
+    `${apiUrl}/bridges/:bridgeId/processors/:processorId`,
+    (req, res, ctx) => {
+      const { bridgeId, processorId } = req.params;
+
+      const existingBridge = db.bridge.findFirst({
+        where: {
+          id: {
+            equals: bridgeId as string,
+          },
+        },
+      });
+
+      if (!existingBridge) {
+        return res(
+          ctx.status(404),
+          ctx.delay(apiDelay),
+          ctx.json({
+            ...error_not_found,
+            reason: `Bridge with id '${
+              bridgeId as string
+            }' for customer 'XXXXXXXX' does not exist`,
+          })
+        );
+      }
+
+      const existingProcessor = db.processor.findFirst({
+        where: {
+          id: {
+            equals: processorId as string,
+          },
+          bridge: {
+            id: {
+              equals: bridgeId as string,
+            },
+          },
+        },
+      });
+
+      if (!existingProcessor) {
+        return res(
+          ctx.status(404),
+          ctx.delay(apiDelay),
+          ctx.json({
+            ...error_not_found,
+            reason: `Processor with id '${
+              bridgeId as string
+            }' for customer 'XXXXXXXX' does not exist`,
+          })
+        );
+      }
+
+      db.processor.update({
+        where: {
+          id: {
+            equals: processorId as string,
+          },
+        },
+        data: {
+          status: "deprovision",
+        },
+      });
+
+      setTimeout(() => {
```

Comment:
See my earlier comment on #25 about the delete process flow (`deprovision`->`deleting`->`deleted`).

Reply:
Done with [cf0c44c](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/30/commits/cf0c44cf68e6801947e0109eba0690a8ccd93925)